### ISSUE TITLE
Segmentation fault on VLA

### ISSUE THREAD

**Issue by iphydf** [esbmc/esbmc#565]
Reproducing test-case:

```c
int a;
int main() {
  char c[a];
  (void)sizeof(c);
}
```

ESBMC version 6.8.0 64-bit x86_64 linux

This also breaks on the web interface.

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: I remember that you did some work on this VLA support. Could you please take a look at this issue?

**Comment by iphydf** [comment]
This is the ubsan stack trace, maybe that helps:

```
../src/util/migrate.cpp:449:32: runtime error: member call on null pointer of type 'namespacet'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/util/migrate.cpp:449:32 in
UndefinedBehaviorSanitizer:DEADLYSIGNAL
==8==ERROR: UndefinedBehaviorSanitizer: SEGV on unknown address 0x000000000000 (pc 0x00000072f8aa bp 0x000000000000 sp 0x7fffb904cad0 T8)
==8==The signal is caused by a READ memory access.
==8==Hint: address points to the zero page.
    #0 0x72f8aa in sym_name_to_symbol(dstring, irep_container<type2t>) /work/esbmc/_build/../src/util/migrate.cpp:449:32
    #1 0x72777a in migrate_expr(exprt const&, irep_container<expr2t>&) /work/esbmc/_build/../src/util/migrate.cpp:714:20
    #2 0x727e7f in migrate_expr(exprt const&, irep_container<expr2t>&) /work/esbmc/_build/../src/util/migrate.cpp:791:5
    #3 0x729ba8 in migrate_expr(exprt const&, irep_container<expr2t>&) /work/esbmc/_build/../src/util/migrate.cpp:1182:5
    #4 0x730547 in splice_expr(exprt const&, irep_container<expr2t>&) /work/esbmc/_build/../src/util/migrate.cpp:430:3
    #5 0x729a7d in migrate_expr(exprt const&, irep_container<expr2t>&) /work/esbmc/_build/../src/util/migrate.cpp:1178:7
    #6 0x725e97 in migrate_type(typet const&) /work/esbmc/_build/../src/util/migrate.cpp:104:7
    #7 0x761734 in c_sizeof(typet const&, namespacet const&) /work/esbmc/_build/../src/util/c_sizeof.cpp:19:15
    #8 0x7c2970 in clang_c_adjust::adjust_sizeof(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:480:20
    #9 0x7c31fc in clang_c_adjust::adjust_operands(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:1189:5
    #10 0x7c2c15 in clang_c_adjust::adjust_expr_binary_arithmetic(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:275:3
    #11 0x7c31fc in clang_c_adjust::adjust_operands(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:1189:5
    #12 0x7c2508 in clang_c_adjust::adjust_expr_rel(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:342:3
    #13 0x7c31fc in clang_c_adjust::adjust_operands(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:1189:5
    #14 0x7ca373 in clang_c_adjust::adjust_ifthenelse(codet&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_code.cpp:69:3
    #15 0x7c31fc in clang_c_adjust::adjust_operands(exprt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:1189:5
    #16 0x7c0ccd in clang_c_adjust::adjust_symbol(symbolt&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:53:5
    #17 0x7c0875 in clang_c_adjust::adjust() /work/esbmc/_build/../src/clang-c-frontend/clang_c_adjust_expr.cpp:44:5
    #18 0x796d97 in clang_c_languaget::typecheck(contextt&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, messaget const&) /work/esbmc/_build/../src/clang-c-frontend/clang_c_language.cpp:229:15
    #19 0x6c8bfb in language_filest::typecheck(contextt&) /work/esbmc/_build/../src/util/language_file.cpp:95:30
    #20 0x6a2c03 in language_uit::typecheck() /work/esbmc/_build/../src/langapi/language_ui.cpp:85:21
    #21 0x48209d in esbmc_parseoptionst::get_goto_program(optionst&, goto_functionst&) /work/esbmc/_build/../src/esbmc/esbmc_parseoptions.cpp:1425:10
    #22 0x47a44e in esbmc_parseoptionst::doit() /work/esbmc/_build/../src/esbmc/esbmc_parseoptions.cpp:467:6
    #23 0x47233e in main /work/esbmc/_build/../src/esbmc/main.cpp:31:23
    #24 0x7f05834660b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #25 0x45066d in _start (/usr/local/bin/esbmc+0x45066d)

UndefinedBehaviorSanitizer can not provide additional info.
SUMMARY: UndefinedBehaviorSanitizer: SEGV /work/esbmc/_build/../src/util/migrate.cpp:449:32 in sym_name_to_symbol(dstring, irep_container<type2t>)
==8==ABORTING
```

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: This VLA example still seems unsupported in ESBMC. It fails with:

```
esbmc: /home/lucas/ESBMC_Project/esbmc/src/util/migrate.cpp:528: expr2tc sym_name_to_symbol(irep_idt, type2tc): Assertion `endatptr != atstr.c_str()' failed.
Aborted
````

Can I ask you please to check this issue?

**Comment by lucasccordeiro** [comment]
ESBMC still fails with this example:

````
ESBMC version 7.2.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing
Converting
esbmc: /home/lucas/ESBMC_Project/esbmc/src/util/migrate.cpp:544: expr2tc sym_name_to_symbol(irep_idt, type2tc): Assertion `endatptr != atstr.c_str()' failed.
Aborted
````

**Comment by lucasccordeiro** [comment]
I assume that @fbrausse is looking at this issue.

**Comment by fbrausse** [comment]
@lucasccordeiro Yes, I am. From duplicate #1290's comment:

> [...] it requires rewriting large parts of ESBMC's frontend(s). There may be other ways to fix this particular issue with the sizeof() of VLAs and I'm looking into one of them. But I wouldn't expect this issue to be fixed soon.

**Comment by lucasccordeiro** [comment]
Thanks, @fbrausse.

**Comment by lucasccordeiro** [comment]
Closed via b24a5ac38b2a86a7d2959b51534e29d263378f92 and ae9af38bec1727f130d500d6c0d366c9b48c132e.
